utils/src/main/kotlin/storage/HttpFileStorage.kt

```diff
@@ -52,6 +52,18 @@ class HttpFileStorage(
      */
     private val cacheMaxAgeInSeconds: Int = 0
 ) : FileStorage {
+    override fun exists(path: String): Boolean {
+        val request = Request.Builder()
+            .headers(headers.toHeaders())
+            .cacheControl(CacheControl.Builder().maxAge(cacheMaxAgeInSeconds, TimeUnit.SECONDS).build())
```

Comment:
Does it make sense to use the cache in an exists check, or should the cache age always be zero?

Reply:
Good question... but given that you probably do an `exists()` just before doing a `read()` or `write()`, I guess you'd want all functions to use the same caching (if any) to avoid getting inconsistent results, or?

Reply:
It might be outdated, but it seems that HEAD requests are not cached anyway:
https://github.com/square/okhttp/issues/2287

Reply:
Hmm. The OkHttp people seem to [suggest](https://github.com/square/okhttp/issues/2287#issuecomment-175699633) to simply do a `GET` without reading the body instead. But even if I don't ready the body in my code, wouldn't that still retrieve the (potentially big) file from the server? I.e. isn't a `HEAD` request still preferable?

Reply:
After reading though https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD I conclud that 1. yes, `HEAD` is preferable over `GET`, and that 2. `HEAD` requests are cachable (as the table says "Cacheable -> Yes").

Reply:
I understood the discussion so that OkHttp does not cache HEAD requests and that they are fine with that because you could use a GET request without reading the body instead. At least in my test with Artifactory earlier today nothing was added to the HTTP cache, so maybe that line could be dropped. But I'll approve and leave that up to you.

Reply:
I'll leave it in for the case that a future OkHttp version starts adding cached `HEAD` requests and to be consistent with the other requests then.